x-pack/plugins/apm/public/components/routing/templates/service_group_template.tsx

```diff
@@ -60,8 +61,19 @@ export function ServiceGroupTemplate({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  const { data: serviceGroupsData, status: serviceGroupsStatus } = useFetcher(
+    (callApmApi) => {
+      if (!serviceGroupId && isServiceGroupsEnabled) {
+        return callApmApi('GET /internal/apm/service-groups');
```

Comment:
do we need to fetch data? I figure we can hide the icon if "all services" is selected, and show it in all other cases?

Reply:
That would be an option to avoid the additional request.

 @formgeist what do you think from the UI point of view? it will look like this 

https://user-images.githubusercontent.com/3369346/165943627-b61b80d7-cb59-4d38-9688-5006f2e52c1a.mov

Reply:
I think it's confusing if the user clicks "All services" group and we still show the "Services" title. cc @boriskirov

Reply:
@kpatticha We could use `isServiceGroupsEnabled` to decide whether to show "Services" or "All services" no?

Reply:
I have fixed the naming in this PR https://github.com/elastic/kibana/pull/131381